docs/developers/content-creation/creating-a-role.md

```diff
@@ -40,3 +44,526 @@ suicideMultiplier = -1
+Now we have to sort out all the players who do not match what we need and when we have found a player who meets our conditions, perform the corresponding function. We can achieve this by using a "for-loop".
+You initialize a for-loop with the following code:
+
+`for i=1, #plys do`
+
+With this, we basically iterate through each player and check them, for the following conditions.
+So next we need a variable that represents the player we are currently checking. we do this by adding the following line:
+
+`local ply = plys[i]`
+
+Now when we access “ply”, we thereby access the player we are checking in the current iteration.
+Next, we want to write a condition that will only be met if the current player is alive and has the role Tester. Feel free to try developing this condition yourself. You can also use the GMOD Wiki if you get stuck. Otherwise you can scroll further and examine the solution.
+
+`if ply:GetSubRole() == ROLE_TESTER and ply:Alive() then`
+
+Next, we use an "if-else statement". Else is used for blocks of code that will be executed instead if the If-statement has not been met.
+With this block we develop the actual function of the role. This statement will define what happens to the tester himself when one of his traitor colleagues has actually died.
+
+Just a reminder of what is supposed to happen: If a fellow traitor dies, the tester's HP should be set to 200 HP if they have 100 or more HP, or set to 125 HP if they have less than 100 HP. 
+First try to develop this block yourself. Once again: Use the GMOD Wiki if there is something unclear. As a little help, I'll show you how to create an if-else nest. If you get stuck, you can scroll down to see the solution:
+
+```lua
+    if [FIRST CONDITION HERE] then
+
+        --[CODE TO PROCESS HERE]
+
+    else 
+
+        --[CODE TO PROCESS HERE]
+
+    end
+```
+
+Once you have set the first condition, due to the else you don't have to develop the second one, because the else will then cover all other cases for you. When you're done, save and test out the role in Garry's mod to see if everything works as planned.
+
+If you did everything right, the code should look something like this:
+
+```lua
+    -- is the player's health beneath 100, he gains an amount of hp
+    if ply:Health() < ply:GetMaxHealth() then
+
+	      ply:SetHealth(125)
+
+    else	 -- is the player's health above 100, he gains another amount of hp
+
+	      ply:SetHealth(200)
+
+    end
+```
+
+To get a better overview, here I have the whole hook for you:
+
+```lua
+if SERVER then
+	
+	hook.Add("PlayerDeath", "ttt2_heal_up_tester", function( victim, inflictor, attacker)
+		if victim:GetSubRole() == ROLE_TESTER then return end
+		if SpecDM and (victim.IsGhost() and victim:IsGhost()) then return end
+		if victim:GetTeam() ~= TEAM_TRAITOR then return end
+		
+		local plys = player.GetAll()
+
+		for i = 1, #plys do
+			local ply = plys[i]
+
+			if ply:GetSubRole() == ROLE_TESTER and ply:Alive() then
```

Comment:
I'm pretty sure you should us `IsActive()` here, because spectators are alive as well